src/Signer.php

```diff
@@ -39,6 +42,39 @@ public function sign($message)
         return $message;
     }
 
+    /**
+     * @param RequestInterface $message
+     * @return RequestInterface
+     */
+    public function signWithDigest($message)
+    {
+        $message = $this->addDigest($message);
+        return $this->sign($message);
+    }
+
+    /**
+     * @param RequestInterface $message
+     * @return RequestInterface
+     */
+    private function addDigest($message)
+    {
+        if (!array_search('digest', $this->headerList->names)) {
+            $this->headerList->names[] = 'digest';
+        };
+        while ($message->getHeader('Digest')) {
```

Comment:
Is this while statement superfluous? Seems `withoutHeader` should return the message without the Digest whether it's present or not?

Reply:
I thought `->withoutHeader()` only remove one instance at a time, but local test confirms otherwise. Removed in 25727bb93fd77099e3cc561cb6ee384cde314ea2